src/scipp/plotting/figure.py

```diff
@@ -78,11 +79,20 @@ def savefig(self, filename=None):
         """
         self.fig.savefig(filename, bbox_inches="tight")
 
-    def _ipython_display_(self):
-        """
-        IPython display representation for Jupyter notebooks.
-        """
-        return self._to_widget()._ipython_display_()
+    if widgets_have_mimebundle():
+
+        def _repr_mimebundle_(self, include=None, exclude=None):
+            """
+            Mimebundle display representation for jupyter notebooks.
+            """
+            return self._to_widget()._repr_mimebundle_(include=include, exclude=exclude)
+    else:
+
+        def _ipython_display_(self):
+            """
+            IPython display representation for Jupyter notebooks.
+            """
+            return self._to_widget()._ipython_display_()
```

Comment:
You could avoid the duplication by using a mixin class: `class Figure(IPythonRepr)` and `IPythonRepr` defines the two functions here. Just a suggestion, it has its own drawbacks...

Reply:
Or use a decorator if you want to avoid inheritance.